homeassistant/components/motion_blinds/config_flow.py

```diff
@@ -138,7 +138,13 @@ async def async_step_connect(self, user_input=None):
             mac_address = motion_gateway.mac
 
             await self.async_set_unique_id(mac_address)
-            self._abort_if_unique_id_configured()
```

Comment:
@bdraco I did not implement DHCP discovery of the Motion Blinds devices since I don't really know how to add DHCP discovery.
How can I figure out the required parameters like the OUI of these gateways?
Is there some python code I can run that prints all discovered devices?

Reply:
Turn on debug logs for `homeassistant.components.dhcp` and power cycle the gateway.

You should get the mac and hostname in the log.

The OUI is just the first 6 of the mac

Reply:
@bdraco I just figured out that the OUI of Motion blinds comes from "Espressif Inc.", I know of 4 gateways all with OUIs from "Espressif Inc.". Unfortunatley all 4 gateways use a diffrent OUI owned by "Espressif Inc." and they own 92 diffrent OUIs, so that is not of much help, that would become a long list....

Reply:
What does the hostname show up as ?

Reply:
> @bdraco I just figured out that the OUI of Motion blinds comes from "Espressif Inc.", I know of 4 gateways all with OUIs from "Espressif Inc.". Unfortunatley all 4 gateways use a diffrent OUI owned by "Espressif Inc." and they own 92 diffrent OUIs, so that is not of much help, that would become a long list....

That's what `registered_devices` is for.  Since you have the mac address in the device entry it will still trigger dhcp flows based on matching that

Reply:
For a Brel blind it shows as "Iwip".
For a Motion gateway is shows as "MOTION_6564"

Reply:
For a separate PR, but something like this should do it

```
diff --git a/homeassistant/components/motion_blinds/config_flow.py b/homeassistant/components/motion_blinds/config_flow.py
index 90ee92ccff..a1119fada4 100644
--- a/homeassistant/components/motion_blinds/config_flow.py
+++ b/homeassistant/components/motion_blinds/config_flow.py
@@ -8,6 +8,7 @@ from homeassistant import config_entries
 from homeassistant.components import network
 from homeassistant.const import CONF_API_KEY, CONF_HOST
 from homeassistant.core import callback
+from homeassistant.components import dhcp
 
 from .const import (
     CONF_INTERFACE,
@@ -72,6 +73,15 @@ class MotionBlindsFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
         """Get the options flow."""
         return OptionsFlowHandler(config_entry)
 
+    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
+        """Handle discovery via dhcp."""
+        mac_address = discovery_info.macaddress  # TODO: reformat if needed
+        await self.async_set_unique_id(mac_address)
+        self._abort_if_unique_id_configured(updates={CONF_HOST: discovery_info.ip})
+
+        self._host = discovery_info.ip
+        return await self._async_handle_discovery()
+
     async def async_step_user(self, user_input=None):
         """Handle a flow initialized by the user."""
         errors = {}
diff --git a/homeassistant/components/motion_blinds/manifest.json b/homeassistant/components/motion_blinds/manifest.json
index adfe7722ca..c218c4cdd1 100644
--- a/homeassistant/components/motion_blinds/manifest.json
+++ b/homeassistant/components/motion_blinds/manifest.json
@@ -5,6 +5,12 @@
   "documentation": "https://www.home-assistant.io/integrations/motion_blinds",
   "requirements": ["motionblinds==0.6.2"],
   "dependencies": ["network"],
+  "dhcp": [
+    {"registered_devices": true},
+    {
+      "hostname": "motion_*"
+    }
+  ],
   "codeowners": ["@starkillerOG"],
   "iot_class": "local_push",
   "loggers": ["motionblinds"]
```

Reply:
The above assumes `_async_handle_discovery` validates the device is a motion blinds device or aborts

Reply:
@bdraco thanks for the example, I will work on this in a seperate PR.
Can you merge this PR?